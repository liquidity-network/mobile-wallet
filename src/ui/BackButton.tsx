import React, { PureComponent } from 'react'
import { TouchableOpacity, ViewStyle, TouchableOpacityProps } from 'react-native'
import Feather from 'react-native-vector-icons/Feather'

import { mediumHitSlop, WHITE } from 'globalStyles'

interface Props {
  onPress: () => void
  iconStyle?: ViewStyle
}

export default class BackButton extends PureComponent<Props & TouchableOpacityProps> {
  render() {
    const { iconStyle, ...props } = this.props
    return (
      <TouchableOpacity activeOpacity={0.7} hitSlop={mediumHitSlop} {...props}>
        <Feather name="chevron-left" style={[iconStyles, iconStyle]} />
      </TouchableOpacity>
    )
  }
}

const iconStyles = { fontSize: 27, color: WHITE, backgroundColor: 'transparent' }
