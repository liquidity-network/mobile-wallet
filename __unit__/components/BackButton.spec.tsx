import React from 'react'
import { create } from 'react-test-renderer'
import 'react-native-vector-icons/Feather'

import BackButton from 'ui/BackButton'

describe('<BackButton> component', () => {
  it('renders correctly', () => {
    const tree = create(<BackButton onPress={() => null} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
