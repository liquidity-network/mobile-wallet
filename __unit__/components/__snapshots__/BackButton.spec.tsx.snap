// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackButton> component renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 15,
      "left": 15,
      "right": 15,
      "top": 15,
    }
  }
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontSize": 27,
          },
          undefined,
        ],
        Object {
          "fontFamily": "Feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï„¯
  </Text>
</View>
`;
